### PR TITLE
handle ssl certificate chains

### PR DESCRIPTION
**Pull request by ddollar**
user can specify --chain for intermediate certs,
otherwise they will be auto-generated

### CODE REVIEW

**Comment by ddollar** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;95gYM0ir&#x2F;235-ssl-certificate-chain-option"> SSL Certificate Chain Option</a>

File: api/models/ssl.go [hunk 1]
```diff
@@ -69,11 +72,22 @@ func CreateSSL(app, process string, port int, body, key string, secure bool) (*S
...
+		chain, err = resolveCertificateChain(body)
+
+		if err != nil {
+			return nil, fmt.Errorf("could not generate chain: %s", err)
+		}
+	}
+
 	name := certName(a.Name, process, port)
 
+	// return nil, fmt.Errorf("foo")
```
Comment (nzoschke) [inline api/models/ssl.go]: Remove commented out code

**Comment by nzoschke** [comment]
The code looks great. :+1: 

Do you need help testing with more inputs?

**Comment by nzoschke** [comment]
While you are making changes here...

https://github.com/convox/rack/issues/185

I see that cmd/convox/ssl.go takes the appFlag. Any reason why .convox/app would cause the appFlag to be ignored?

**Comment by nzoschke** [comment]
Finally, can you submit a documentation change for this?

Also while we are making changes here... based on @asyazwan feedback in Slack I also noticed we don't have documentation around the `--secure` option to SSL.

**Comment by ddollar** [comment]
Docs change is convox/convox.github.io#75

**Pull request merged**